cli/azd/pkg/apphost/aca_ingress.go

```diff
@@ -0,0 +1,172 @@
+package apphost
+
+import (
+	"fmt"
+)
+
+// buildAcaIngress builds the Azure Container Apps ingress configuration from the provided bindings.
+func buildAcaIngress(bindings map[string]*Binding, defaultIngressPort int) (*genContainerAppIngress, error) {
+	if len(bindings) == 0 {
+		return nil, nil
+	}
+
+	if err := validateInput(bindings); err != nil {
+		return nil, err
+	}
+
+	endpointByContainerPort := make(map[string][]*Binding)
+	for _, binding := range bindings {
+		bindingKey := "default" // default is for those with no container port
+		if binding.TargetPort != nil {
+			bindingKey = fmt.Sprintf("%d", *binding.TargetPort)
+		}
+		endpointByContainerPort[bindingKey] = append(endpointByContainerPort[bindingKey], binding)
+	}
+
+	endpointByContainerPortProperties := make(map[string]*endpointGroupProperties)
+	for containerPort, bindings := range endpointByContainerPort {
+		props := &endpointGroupProperties{httpOnly: true}
+		for _, binding := range bindings {
+			if binding.TargetPort != nil {
+				props.port = *binding.TargetPort
+			}
+			if binding.External {
+				props.external = true
+			}
+			if binding.Scheme != acaIngressSchemaHttp && binding.Scheme != acaIngressSchemaHttps {
+				props.httpOnly = false
+			}
+			if binding.Transport == acaIngressTransportHttp2 {
+				props.hasHttp2 = true
+			}
+		}
+		endpointByContainerPortProperties[containerPort] = props
+	}
+
+	countExternalGroups := 0
+	for _, props := range endpointByContainerPortProperties {
+		if props.external {
+			countExternalGroups++
+		}
+		if props.external && !props.httpOnly {
+			return nil, fmt.Errorf("External non-HTTP(s) endpoints are not supported")
+		}
+	}
+	if countExternalGroups > 1 {
+		return nil, fmt.Errorf("Multiple external endpoints are not supported")
+	}
+
+	var httpOnlyGroups []string
+	for groupKey, props := range endpointByContainerPortProperties {
+		if props.httpOnly {
+			httpOnlyGroups = append(httpOnlyGroups, groupKey)
+		}
+	}
+
+	var ingress string
+	if len(httpOnlyGroups) == 1 {
+		ingress = httpOnlyGroups[0]
+	}
+
+	if ingress == "" {
+		// We have more than one, pick prefer external one
+		var externalHttpOnly []string
+		for _, groupKey := range httpOnlyGroups {
+			if endpointByContainerPortProperties[groupKey].external {
+				externalHttpOnly = append(externalHttpOnly, groupKey)
+			}
+		}
+
+		if len(externalHttpOnly) == 1 {
+			ingress = externalHttpOnly[0]
+		} else if len(httpOnlyGroups) > 1 {
+			return nil, fmt.Errorf("Multiple internal only HTTP(s) endpoints are not supported")
+		}
+	}
+
+	additionalPortsCount := len(endpointByContainerPort)
+	if ingress != "" {
+		additionalPortsCount--
+	}
+	if additionalPortsCount > 5 {
+		return nil, fmt.Errorf("More than 5 additional ports are not supported. " +
+			"See https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview#tcp for more details.")
```

Comment:
Since I read the test code first, I commented on the `en-us` bit there, but here it is in the product, if this helps you find it faster and remove it :-)